### PR TITLE
feat(build): support branches build

### PR DESCRIPTION
**Pull request by rbioteau**
What has been done:

- [x] Update `BONITA_VERSION` variable to `BRANCH_OR_TAG`
- [x] Verify if the given git ref is either a branch or a tag
- [x] Update the maven wrapper to latest version on the different project to support `-ntp` option
- [x] Fix `bonita-web-pages` ui designer version 
- [x] Fix `bonita-web-pages` repositories (add `mavenLocal()`)  https://github.com/bonitasoft/bonita-web-pages/commit/0ef06e0d631527443e5c9d1e54a1cb7a1de03e5e
- [x] Update failure slack notification
- [x] Add a cron to run the build each night
- [x] Set `BRANCH_OR_TAG` to `dev` branch by default

> :bookmark: **TODO** Remove the `dev` branch and rename `master` into `main`. As community doesn't have maintenance version aymore, there is no need for `dev`+ `master` branch.

Fixes https://github.com/Bonitasoft-Community/Build-Bonita/issues/41

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -30,11 +32,13 @@ jobs:
...
     - name: Checkout
-      uses: actions/checkout@v1
+      uses: actions/checkout@v3
+      
     - name: Setup JDK
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
       with:
         java-version: 11
+        distribution: 'temurin'
```
Comment (tbouffard) [inline .github/workflows/ci.yml:41]: ❓ is it a reason to force this distribution?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -44,11 +48,11 @@ jobs:
       shell: bash
       run: ./build-script.sh
       env:
-        BONITA_BUILD_QUIET: true
+        BONITA_BUILD_QUIET: false
```
Comment (tbouffard) [inline .github/workflows/ci.yml:51]: ❓ is it done on purpose or a remaining setting used to temporary debug?
Setting this to true highly increase the log size. In the past, it didn't help a lot to debug issue when build errors occurred. But maybe this is now more useful.
Same question for the Studio part.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -44,11 +48,11 @@ jobs:
       shell: bash
```
Comment (tbouffard) [inline .github/workflows/ci.yml:48]: ℹ️ let's rename the step to `Build bundle`

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -57,47 +61,23 @@ jobs:
...
+      uses: actions/upload-artifact@v3
       with:
         name: bonita-studio-${{matrix.os.short}}-build-${{github.sha}}
         path: bonita-studio/all-in-one/target/*.zip
-    - name: Send a message to Slack channel
-      if: ${{ failure()}}
-      uses: slackapi/slack-github-action@v1.17.0
+    
+    - name: Send failure message to Slack channel
+      if: ${{  github.event.number == 0 && failure() }}
```
Comment (tbouffard) [inline .github/workflows/ci.yml:74]: 👍🏿 nice, slack notification for PR was spammy!

Can we use the `github.event_name != 'pull_request'` condition instead? He makes it clear that we do not consider PR. `github.event.number == 0` works but is less explicit IMHO.

**Review by tbouffard — COMMENTED** [review]
I don't have time to review the changes in details. Can @educhastenier have a look at it?

This sounds good and is a big step forward. 🍾

File: .github/workflows/ci.yml (hunk 1 above)
Comment (rbioteau) [reply .github/workflows/ci.yml:41]: It is [mandatory to set a distribution](https://github.com/actions/setup-java#usage).

_Temurin_ is just a personal preference as an Eclipse Foundation enthousiat :stuck_out_tongue:

File: .github/workflows/ci.yml (hunk 2 above)
Comment (rbioteau) [reply .github/workflows/ci.yml:51]: I have added the `-ntp`options for maven builds. It removes the transfer progress logs. Current PR logs is 2 Mo.

Comment (tbouffard) [reply .github/workflows/ci.yml:51]: ok fine, thanks for the details

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tbouffard) [reply .github/workflows/ci.yml:41]: yes you are right, I have noticed that you bump the java actions 😸

**Pull request merged**